lib/pubsubstub/subscription.rb

```diff
@@ -0,0 +1,63 @@
+module Pubsubstub
+  class Subscription
+    include Logging
+
+    attr_reader :channels, :connection, :queue, :id
+
+    def initialize(channel_names, connection)
+      @id = Random.rand(2 ** 64)
+      @connection = connection
+      @channels = channel_names.map { |n| Channel.new(n) }
+      @queue = Queue.new
+    end
+
+    def stream(last_event_id)
+      info { "Connecting client ##{id} (#{channels.map(&:name).join(', ')})" }
+      fetch_scrollback(last_event_id)
+      subscribe
+      while event = queue.pop
```

Comment:
Doesn't this break EM installs by stalling the event loop? If so, then the `StreamAction` should at least bail after giving the scrollback